cmd/quotas.go

```diff
@@ -130,11 +131,106 @@ var quotasCmd = &cobra.Command{
 	},
 }
 
+var defaultResources = corev1.ResourceList{
+	// CPU
+	"requests.cpu": resource.MustParse("70"),
+	"limits.cpu":   resource.MustParse("70"),
+
+	// Memory
+	"requests.memory": resource.MustParse("368G"),
+	"limits.memory":   resource.MustParse("368G"),
+
+	// Storage
+	"requests.storage": resource.MustParse("4T"),
+
+	// GPU
+	"requests.nvidia.com/gpu": resource.MustParse("2"),
+
+	// Pods
+	"pods": resource.MustParse("100"),
+
+	// Services
+	"services.nodeports":     resource.MustParse("0"),
+	"services.loadbalancers": resource.MustParse("0"),
+}
+
+var quotaLabels = [9]string{
+	"quotas.statcan.gc.ca/requests.cpu", "quotas.statcan.gc.ca/limits.cpu",
+	"quotas.statcan.gc.ca/requests.memory", "quotas.statcan.gc.ca/limits.memory",
+	"quotas.statcan.gc.ca/requests.storage", "quotas.statcan.gc.ca/gpu",
+	"quotas.statcan.gc.ca/pods", "quotas.statcan.gc.ca/services.nodeports",
+	"quotas.statcan.gc.ca/services.loadbalancers",
+}
+
+var quotaPrefixLabel = "quotas.statcan.gc.ca/"
+
+func hasQuotaLabel(profileLabel string, key string, profile *kubeflowv1.Profile) bool {
+	if _, ok := profile.Labels[profileLabel]; ok {
+		if strings.HasPrefix(profileLabel, quotaPrefixLabel) {
+			s := strings.TrimPrefix(profileLabel, quotaPrefixLabel)
+			if s == key {
+				return true
+			}
+			return false
+		}
+		return false
+	}
+	return false
+}
+
+func overrideResourceQuotas(profile *kubeflowv1.Profile) {
+
+	for key, _ := range defaultResources {
+		// CPU
+		if hasQuotaLabel(quotaLabels[0], key.String(), profile) {
+			cpuRequest := profile.Labels[quotaLabels[0]]
+			defaultResources[key] = resource.MustParse(cpuRequest)
+		}
+		if hasQuotaLabel(quotaLabels[1], key.String(), profile) {
+			cpuLimit := profile.Labels[quotaLabels[1]]
+			defaultResources[key] = resource.MustParse(cpuLimit)
+		}
+		// Memory
+		if hasQuotaLabel(quotaLabels[2], key.String(), profile) {
+			memoryRequest := profile.Labels[quotaLabels[2]]
+			defaultResources[key] = resource.MustParse(memoryRequest + "G")
```

Comment:
Instead of adding `G` lets just say users have to say "quotas.statcan.gc.ca/requests.memory: 100Gi", and not assume the unit